lib/vrl/stdlib/src/parse_nginx_log.rs

```diff
@@ -0,0 +1,225 @@
+use crate::log_util;
+use regex::Regex;
+use std::collections::BTreeMap;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseNginxLog;
+
+impl Function for ParseNginxLog {
+    fn identifier(&self) -> &'static str {
+        "parse_nginx_log"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "format",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "timestamp_format",
+                kind: kind::BYTES,
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let variants = vec![value!("combined"), value!("error")];
+
+        let value = arguments.required("value");
+        let format = arguments
+            .required_enum("format", &variants)?
+            .try_bytes()
+            .expect("format not bytes");
```

Comment:
I would prefer to use:

```rust
.try_bytes()?;
```

rather than `expect` here. Although it should never happen, since the compiler ensures an error occurs if the parameter is not a bytes before we get to this line, should something go wrong somehow it would be better for the user to get an error message rather than a panic.

(I realise there are other functions that aren't completely consistent here.)

Reply:
No problem, it's because it was in `parse_apache_log` but I agree with you ;)

Reply:
Also, this implies more development: `try_bytes`'s error comes from `lib/vrl/compiler/src/value/error.rs:Error` and `Compiled` (returned by this function) is `Box<dyn DiagnosticError>`.
And I'm not sur this should be done in this PR 🤔

Reply:
Oh, interesting. Yes you are right. There should be an `Into` implemented to move between them, but there doesn't seem to be. So it will have to be like this.

It shouldn't be an issue however, since it should never error if it gets to this stage.